### ISSUE TITLE
NUnitLite - Explore flag does not apply where filter to output

### ISSUE THREAD

**Issue by chriscore** [nunit/nunit#1143]
Hi there

Running nunit3-console.exe with arguments similar to the following does not apply the filter specified in /where to the results which are seen on the output.

"C:\MyRepo\MyAssembly.dll" /explore /noheader /where="cat==Awesome"

All tests are returned by /explore, including those which do not satisfy cat==Awesome

This was found in NUnit Console Runner v3.0.5813

**Comment by CharliePoole** [comment]
I was uncertain whether it made sense to use a filter when listing tests, but I suppose it does if you tried it!

**Comment by chriscore** [comment]
Hey Charlie

Thanks, before a test run I am currently trying to call this to see how many tests my filter would return - it would certainly be very useful for me at this point :)

Great work with nunit 3 so far

Chris

**Comment by CharliePoole** [comment]
@chriscore If you have a runner, just call CountTestCases on it with your filter!

If you are simply scripting nunit3-console, that's not an option, of course.

**Comment by chriscore** [comment]
@CharliePoole I don't currently have a runner instance as I haven't been able to figure out how to get that going, like you say I have just scripted nunit3-console, but having that control in process would be ideal.

I did find it difficult to find how to host a runner, do you have a C# snippet anywhere which might give some clues how to start? Being able to load and run tests in process would be ideal for what I'm trying to do here.

**Comment by CharliePoole** [comment]
Add a reference to nunit.engine.api.dll. Then (simplified)...

```
ITestEngine engine = TestEngineActivator.CreateInstance();
TestPackage package = new TestPackage(ASSEMBLY_PATH);
ITestRunner runner = engine.GetRunner(package);
```

This leaves out setting options, using a filter, etc. For the full story, start here:
    https://github.com/nunit/nunit/blob/master/src/NUnitConsole/nunit3-console/Program.cs#L111

**Comment by chriscore** [comment]
Thanks so much for that, I didn't know about the TestEngineActivator!

**Comment by rprouse** [comment]
When this is fixed, ensure it also works with NUnitLite. See #1213

**Comment by NCarter3** [comment]
I'd also love to see filters working with --explore!

**Comment by ChrisMaddock** [comment]
Copying this issue to the console repo, and keeping this one for NUnitLite

**Comment by CharliePoole** [comment]
There are two parts to this: a runner has to pass the filter info to the framework and the framework has to pay attention to it. In this case the runner is nunitlite. The corresponding console issue nunit/nunit-console#139 can't really be worked on until at least the framework part of this one is completed.

**Comment by CharliePoole** [comment]
Travis doesn't seem to be able to find Apple agents tonight, but the Linux jobs all passed so I'm going to go ahead and merge.

**Comment by Poimen** [comment]
@ChrisMaddock @CharliePoole As per our brief discussion on the nunit-discussion forum, I'll be happy to have a look into this issue for you.

**Comment by rprouse** [comment]
Closed by PR #2094
